### DISCUSSION TITLE
convert ideacon file to xml

### DISCUSSION THREAD

**Discussion by mohammed-mohammed-m** [idea-statica/ideastatica-public#181] [Help Q&A] [answered]
Hi, I have a lot of connections to make drawing from, so I need a code to convert Idea StatiCa file *.ideacon to *.xml or *.IFC to import it in Revit can you help me with that!

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi Mohammed, 

Unfortunately, we do not currently provide a direct conversion to Revit or IFC. Although, it is something we are actively considering. 

You can export a connection to IOM .xml via the developer tab in IDEA Connection. To get access to the developer tab see this [page](https://github.com/idea-statica/ideastatica-public/wiki/Developer-Tools-in-IDEA-StatiCa-Connection)

This will export the connection file to its IOM representation and will contain an open geometric representation of the plate, weld, bolt geometry in IOM format. You won’t be able to directly import this into Revit though without converting it somehow using the Revit Connection API. 

Alternatively, you can try using the Online Connection Viewer to Export the Connections to .DXF and try and reference these into Revit.
https://www.ideastatica.com/support-center/missing-components-in-3d-dwg-export-from-viewer

Out of interest are you using Checkbot here to manage your connections or are they all in separate IDEA Connection files?

**Reply by mohammed-mohammed-m** [reply]
Hi Nathan,

Thanks for support, hope next version have export to different file types to be easier to make construction drawings,
The Viewer is excellent but when it comes to tube sections it will be divided to segments, but if there .xml or IFC then all date is stored in detailed properties then Revit can import it as connection (through IFC file), now I'm using SketchUp Pro to convert dwg file from the Viewer to IFC then import it in Revit.
I didn't used Checkbot yet but I'll try it soon, I have separate IDEA Connection files.

**Reply by nathanlukeIDEA** [reply]
Thanks, Mohammed, 

To confirm when you say '.xml' do you mean an XML representation of IFC?

**Reply by mohammed-mohammed-m** [reply]
Hi Nathan,
Yes, because I want to use it for detailing and I think a lot of people like me want easy way to make use of analyzed connection to make detail of by using Revit it is easier than AutoCAD in detailing, 
sorry for being late...

**Reply by tsoumdoa** [reply]
Is anyone actively trying to develop this IOM to Revit/ IFC workflow? I'm looking into developing something for work since the current dxf export is far from ideal...

**Reply by nathanlukeIDEA** [reply]
Hi @tsoumdoa, we are working towards an IFC export. Although, unfortunately, I can't say now exactly when it will be available.

**Reply by Balkmannen** [reply]
In IdeaRS.OpenModel you can use "XaveToXmlFile". Is it possible to open an .ideacon-file in OpenModel and use that command to create the XML? Or is it possible to open the ideacon-file with the ConnHiddenClientFactory.Create().OpenProject('ideacon-file') and save to xml from there?

**Reply by Balkmannen** [reply]
Never mind. I managed to find the information like this:

ideaConnectionClient.OpenProject(connection_project_path)
                projectInfo = ideaConnectionClient.GetProjectInfo()
                connections = projectInfo.Connections
                firstCon = connections[0]
                my_id = firstCon.Identifier
                print(ideaConnectionClient.GetAllConnectionData(my_id))
